src/eth.rs

```diff
@@ -241,11 +241,15 @@ impl RxRing {
     fn buf_release(&mut self) {
         let addr = &self.pkt_buf[self.cur_desc] as *const _;
         self.desc_buf[self.cur_desc][0] = addr as u32 & EMAC_DES0_BUF1AP;
+        // Ensure change to desc0 occurs before setting desc3 OWN flag
+        cortex_m::asm::dmb();
```

Comment:
Not sure this is required as long as the tail pointer is only updated once both are written.

Reply:
Yep - you are right.